pennylane/data/dataset.py

```diff
@@ -151,5 +149,54 @@ def from_dataset(cls, dataset):
             >>> print(vars(new_dataset))
             {'dtype': None, '__doc__': '', 'kw1': 1, 'kw2': '2', 'kw3': [3]}
         """
-        kwargs = {key: val for (key, val) in dataset.__dict__.items() if key not in ["dtype"]}
-        return cls(dtype=dataset.dtype, **kwargs)
+        return cls(**dataset.attrs)
+
+
+class RemoteDataset(Dataset):
```

Comment:
Why we need this class?

Reply:
The Dataset class is generic, and should be the base. Users can use it directly like `Dataset(my_attr=my_val, other_attr=other_val)` with any attributes they want. However, our datasets have uniformity in how they're stored (via `qml.data.load`), and this class has intelligence based on the uniformity we maintain in the remote datastore (S3 bucket). It's an expansion on the `Dataset` base class with opinionated intelligence on where data files should be. The naming can definitely be changed, it's just what I thought of at the time. Plus, I assume the user will first interact with this library in one of two ways:
1. `qml.data.Dataset(**some_args)`, to create their own custom data
2. `qml.data.load(description_of_our_data_they_want)`, to get a list of Dataset objects. In this case, the type will actually be a "RemoteDataset" (because they got it from a remote location), but it will do the same stuff and they'll never even have to type out "RemoteDataset" or even know that it distinctly exists! (unless of course they read the docs we'll provide on it 😋)

Reply:
Is it possible to merge the two class to have a general one that fulfils our requirements for finding and loading the data?
Maybe we can default the `arguments` that manage loading the PL data to `None` and pass them in the `load` function?

Maybe I am not getting the logic, but I do not see any good reason for having two `Dataset` classes.

Reply:
in my opinion, after trying to implement this more than once, no. We can make convoluted meta-classes and hidden fields that are useless when configured in another way, but I found it to be a worse experience in general. To name one example problem, Python doesn't support constructor overloading in any friendly way. This makes it hard to give a uniform experience to all users. In my mind, the RemoteDataset is a strict superset of Dataset, and this puts its functionality in a nice home.

I considered making the base `Dataset` class abstract (and have two implementations) to be more intentional with this design choice, but it didn't make sense because the user might update a `RemoteDataset` to have custom user data (which is allowed! It's a Dataset, after all).

tl;dr: I'm ok to normalize to one class if we really want it, but I caution that the code would be full of anti-patterns and much harder to read.

Reply:
I'm trying to squash them into a single class one last time rn, might be able to make it work without causing a mess. Otherwise an appropriate renaming will definitely happen before merging